helm-charts/splunk-otel-collector/templates/config/_otel-agent.tpl

```diff
@@ -75,6 +75,25 @@ receivers:
           endpoint: '`endpoint`:`"prometheus.io/port" in annotations ? annotations["prometheus.io/port"] : 9090`'
       {{- end }}
 
+      # Receivers for collecting k8s control plane metrics.
+      {{- if .Values.autodetect.controlPlane }}
+      smartagent/kubernetes-apiserver:
+        {{- $rule_suffix :=  "port == 443 && pod.labels[\"k8s-app\"] == \"kube-apiserver\"" -}}
+        {{- if eq .Values.distribution "openshift" }}
+        {{- $rule_suffix =  "port == 6443 && pod.labels[\"app\"] == \"openshift-kube-apiserver\"  && pod.labels[\"apiserver\"] == \"true\"" -}}
+        {{- end }}
+        rule: type == "port" && {{ $rule_suffix }}
+        config:
+          discoveryRule: target == "port" && {{ $rule_suffix }}
```

Comment:
you will still need to use `rule` - `discoveryRule` is from the signalfx-agent service discovery.

Reply:
sorry didnt see the rule right above this. just remove this line completely.

Reply:
Now I know how that works. Done.